From d05cdd1c052d8c6b746a3c384e553b049dac3f41 Mon Sep 17 00:00:00 2001
From: Bet4 <0xbet4@gmail.com>
Date: Tue, 28 Jul 2026 19:53:17 +0800
Subject: [PATCH] Bypass systemd-run in nixos-rebuild-ng

Asterinas does not currently provide systemd's D-Bus interface, so nixos-rebuild-ng cannot use its systemd-run wrapper.

TODO: Remove this patch once systemd D-Bus support is implemented.
---
 nixos_rebuild/nix.py | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/nixos_rebuild/nix.py b/nixos_rebuild/nix.py
index c6b5b4d63c45..eff7a1548865 100644
--- a/nixos_rebuild/nix.py
+++ b/nixos_rebuild/nix.py
@@ -31,21 +31,21 @@ from .utils import Args, dict_to_flags
 FLAKE_FLAGS: Final = ["--extra-experimental-features", "nix-command flakes"]
 FLAKE_REPL_TEMPLATE: Final = "repl.nix.template"
 SWITCH_TO_CONFIGURATION_CMD_PREFIX: Final = [
-    "systemd-run",
-    "-E",
-    # Will be set to new value early in switch-to-configuration script,
-    # but interpreter starts out with old value
-    "LOCALE_ARCHIVE",
-    "-E",
-    "NIXOS_INSTALL_BOOTLOADER",
-    "-E",
-    "NIXOS_NO_CHECK",
-    "--collect",
-    "--no-ask-password",
-    "--pipe",
-    "--quiet",
-    "--service-type=exec",
-    "--unit=nixos-rebuild-switch-to-configuration",
+    # "systemd-run",
+    # "-E",
+    # # Will be set to new value early in switch-to-configuration script,
+    # # but interpreter starts out with old value
+    # "LOCALE_ARCHIVE",
+    # "-E",
+    # "NIXOS_INSTALL_BOOTLOADER",
+    # "-E",
+    # "NIXOS_NO_CHECK",
+    # "--collect",
+    # "--no-ask-password",
+    # "--pipe",
+    # "--quiet",
+    # "--service-type=exec",
+    # "--unit=nixos-rebuild-switch-to-configuration",
 ]
 logger: Final = logging.getLogger(__name__)
 
-- 
2.54.0

